### ISSUE TITLE
local_action / delegate_to: 127.0.0.1 are trying to use SSH no matter what I do

### ISSUE THREAD

**Issue by enchantner** [ansible/ansible#17952]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
number.py:57: PowmInsecureWarning: Not using mpz_powm_sec.  You should rebuild using libgmp >= 5 to avoid timing attack vulnerability.
  _warn("Not using mpz_powm_sec.  You should rebuild using libgmp >= 5 to avoid timing attack vulnerability.", PowmInsecureWarning)
ansible 2.1.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

My ansible.cfg looks like this:
[default]
force_handlers=True
[ssh_connection]
pipelining=True

There are no ANSIBLE_\* variables overridden.
##### OS / ENVIRONMENT

Centos 6.5
##### SUMMARY

I have an Ansible playbook what I'm trying to run to configure several nodes. At some point, I'm trying to run a shell command on a local host.

The fun thing is, it doesn't matter if I use "local_action" or "delegate_to: 127.0.0.1", it still tries to backconnect to a host node from target nodes - I see lines like "failed: [1.1.1.1 -> localhost]" in logs.

After googling, I suggested to check an 'ansible_connection' variable, and it's set to 'smart', which is by default. 
##### STEPS TO REPRODUCE

```
    - name: Local action
      run_once: true
      become: false
      local_action: shell "blablabla"
```

##### EXPECTED RESULTS

Command is running on localhost
##### ACTUAL RESULTS

Ansible tries to connect from target nodes to host node

```

```

**Comment by jctanner** [comment]
@enchantner Please provide the ansible-playbook -vvvv output for the delegated task. I would also like to see the following:
1. The full ansible-playbook command line options used
2. The inventory file
3. The playbook header (hosts/gather_facts/connection/etc)

needs_info

**Comment by Provscbunn** [comment]
This is happening for me with winrm.  delegate_to, local_action both result in trying to make a winrm connection to the localhost.  Why would either of these actions try and make a connection anywhere?

**Comment by 2rs2ts** [comment]
@enchantner you might be running into the same issue we had where the working directory of the `command` task wasn't the same anymore due to this bug fix: https://github.com/ansible/ansible/commit/9255a618e307b4e087cb0f3f6f4f225f7652c298

You might want to see if that's the problem by using `chdir` if you're referring to a command via a relative path.

**Comment by ansibot** [comment]
@enchantner ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@enchantner ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@enchantner This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@enchantner This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@enchantner You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
